compiler/src/iree/compiler/Dialect/Util/IR/IntegerDivisibility.h

```diff
@@ -0,0 +1,100 @@
+// Copyright 2024 The IREE Authors
```

Comment:
merge into UtilTypes.h instead of forward declaring - trying to keep the IR/ dirs concise and one fewer file is net equal to hiding the simple types compared to the atrocities that the tablegen outputs produce

Reply:
Yeah. Sorry - was fiddling with trying to keep from pulling in deps like raw_ostream and a prior iteration needed the Lattice class from the analysis framework. Was trying to sequester the header deps to avoid everyone needing to pull them in to have access to the interface. But I think it is down to raw_ostream now, which I guess can be forward declared.

Reply:
I guess I could move the `operator<<` functions to the analysis header if I have to, but the framework does require that these types can be output to a raw_ostream.

Reply:
I pull it in in other util files, so anyone depending on it will get it (UtilInterfaces.td needs it)

Reply:
SparseAnalysis would be the bigger one - but I'm thinking more that this kind of thing doesn't belong on ops if it's pulling all that in - it should be in Analysis/ - we'd still need IR/ to dep on the analysis stuff this way

Reply:
(feels like analysis should be operating on ops, instead of mixed in with their definitions)

Reply:
Upstream Analysis depends on the global interfaces. It's just a narrow path that is easy to fall off. I had the lattice itself in the interfaces side to start with and that is where the badness happened.